ts-secrets/src/secret_retrieval.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * @klotho::execution_unit {
+ *  id = "secrets-retriever"
+ * }
+ */
+
+
+/**
+ * @klotho::persist {
+ *  secret = true
+ * id = "secret"
+ * }
+ */
+ import fs = require("fs/promises");
+
+
+export async function getSecret(): Promise<Buffer> {
+      // NOTE: after klo-compile, copy this 'my_secret.key' into the compiled directory
+  // and the `pulumi up` will upload the secret for you.
```

Comment:
```suggestion
  // NOTE: after klo-compile, copy this 'my_secret.key' into the compiled directory
  // and the `pulumi up` will upload the secret for you.
```

Reply:
not sure we need it to be ts-secrets, but what do we want it to be then? its weird to just have a sample solely for proxy since its really not different than microservices

Reply:
we could make our tests take in multiple configs to test but that seems more complicated

Reply:
> its weird to just have a sample solely for proxy since its really not different than microservices

Yeah, how _is_ it different from microservices? Is the intersection of proxy + secrets special somehow?

I think you may have some context about the reason for this PR that I lack. :)

Reply:
oh the reason is we dont test lambda proxy anywhere which is why we didnt catch that our latest release doesnt work for lambda proxying

my hope with this was just to make sure its tested while we figure out a better way to do it. I cut https://github.com/CloudCompilers/klotho/issues/530 for that

Reply:
huh, ts-microservices doesn't test lambda proxy? Oh, is that because we explicitly created microsrv-users as fargate?

I think the way to do this is via multiple configs. If we don't want to go down that route, I would argue for essentially a copy of ts-microservies: `ts-microservices-ldambda` or such — with a note on #503 to rm that sample app when we do a better way.

Shoehorning the proxy into some random sample app seems not-great and confusing.

Reply:
ok ill create lambda microservices then, im fine with that

Reply:
Ah this is also because we currently only test 1 variation of the klotho.yaml config file right? Would this be grounds to update that and allow testing of different klotho yaml files so we don't need to keep making new samples?

Reply:
yeah i think thats the long term solution we should take to solve https://github.com/CloudCompilers/klotho/issues/530, but for now i want to make sure we at least test lambda -> lambda since it will be common